client/src/test/scala/slack/api/MockSttpBackend.scala

```diff
@@ -0,0 +1,19 @@
+package slack.api
+
+import com.github.dapperware.slack.AccessToken
+import sttp.client3.testing.SttpBackendStub
+import zio.Task
+import sttp.client3.asynchttpclient.zio.AsyncHttpClientZioBackend
+import sttp.capabilities.WebSockets
+import sttp.capabilities.zio.ZioStreams
+import zio.{ Layer, ZLayer }
+import sttp.client3.asynchttpclient.zio.SttpClient
+
+trait MockSttpBackend {
+  def sttpBackEndStub: SttpBackendStub[Task, ZioStreams with WebSockets] =
+    AsyncHttpClientZioBackend.stub
+
+  def accessTokenLayer(accessToken: String) = AccessToken.make(accessToken).toLayer
+
+  def sttpClientLayer: Layer[Nothing, SttpClient] = ZLayer.fromFunction((_: Any) => sttpBackEndStub)
```

Comment:
There should be a `AsyncHttpClientZioBackend.stubLayer` that you can use instead, then you can set up the mocks by doing a before block.

```scala
@@ before(stub.whenRequestMatches {
 /// ...
})
```

Reply:
1. Setting up mocks with a stub Layer would look something like this
```
def sttpBackEndLayer: Layer[Throwable,SttpClient with SttpClientStubbing] =
    AsyncHttpClientZioBackend.stubLayer.map{ env =>
        val stub = env.get[SttpClientStubbing.Service]
        stub.whenRequestMatchesPartial {
            case req if req.uri.toString == "some-uri" => Response.apply("positive-response", StatusCode.Ok)
        }
     ??? 
    }
```
This looks very tedious when compared to the current approach. Is there a better way of using `AsyncHttpClientZioBackend.stubLayer` ?

2. Re the before block, I am assuming you are talking about overriding `beforeEach`. If so, the sttp backend stub does not have an api to forget the previous mocks.

Reply:
I gave it a shot (used `stubLayer` instead of stub). Please let me know if I have incorporated your feedback in that.